lib/siwapp/invoices/invoice.ex

```diff
@@ -88,6 +89,72 @@ defmodule Siwapp.Invoices.Invoice do
     |> validate_length(:email, max: 100)
     |> validate_length(:contact_person, max: 100)
     |> validate_length(:currency, max: 100)
+    |> maybe_set_amounts()
+  end
+
+  def maybe_set_amounts(changeset) do
+    if are_items_set?(changeset) do
+      changeset
+      |> set_net_amount()
+      |> set_gross_amount()
+    else
+      changeset
+    end
+  end
+
+  defp set_net_amount(changeset) do
+    net_amount =
+      changeset
+      |> calculate_net_amount()
+
+    put_change(changeset, :net_amount, net_amount)
+  end
+
+  defp set_gross_amount(changeset) do
+    taxes_amount =
+      changeset
+      |> calculate_taxes_amount()
+
+    net_amount = get_field(changeset, :net_amount)
+
+    put_change(changeset, :gross_amount, net_amount + taxes_amount)
+  end
+
+  defp calculate_net_amount(changeset) do
+    item_changeset =
+      changeset
+      |> get_field(:items)
+
+    net_amount = for val <- item_changeset, do: val.net_amount
+
+    Enum.sum(net_amount)
```

Comment:
En vez del for anterior, no podrías mejor:
`Enum.map(item_changeset, &(&1.net_amount))
|> Enum.sum()
|> round()`

Asi todo junto?